ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/browser/BrowserViewFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Project: RCS - Rail Control System
+ *
+ * © Copyright by SBB AG, Alle Rechte vorbehalten
+ */
+package ch.sbb.scion.rcp.microfrontend.browser;
+
+import static com.teamdev.jxbrowser.engine.RenderingMode.HARDWARE_ACCELERATED;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+
+import com.teamdev.jxbrowser.engine.Engine;
+import com.teamdev.jxbrowser.engine.EngineOptions;
+import com.teamdev.jxbrowser.view.swt.BrowserView;
+
+/**
+ * Factory class for creating different types of {@link BrowserView} instances. This class provides methods to create browser views for both
+ * JxBrowser and SWT environments.
+ */
+public final class BrowserViewFactory {
+
+  /**
+   * Creates a new {@link BrowserView} instance of the specified type within the provided composite.
+   *
+   * @param type
+   *          The type of the browser view to create.
+   * @param composite
+   *          The parent composite in which the browser view will be embedded. Must not be {@code null}.
+   * @return A new {@link BrowserView} instance created with the specified type and embedded in the given composite.
+   */
+  public static AbstractBrowserView createBrowserView(final BrowserViewType type, final Composite composite) {
```

Comment:
We do have an interface for `AbstractBrowserView`, right? And we say so in our own docs:

> Factory class for creating different types of {@link BrowserView} instances.

If so, why are we returning the abstract type?